internal/clientcache/internal/daemon/status_handler.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/clientcache/internal/cache"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/util"
+)
+
+// RefreshTokenStatus is the status of a resource token
+type RefreshTokenStatus struct {
+	Age      time.Duration `json:"age"`
+	LastUsed time.Duration `json:"last_used"`
+}
+
+type ErrorStatus struct {
+	Error        string        `json:"error"`
+	LastReturned time.Duration `json:"last_returned"`
+}
+
+// ResourceStatus contains the status of a specific resource type contained in
+// the cache for a specific user.
+type ResourceStatus struct {
+	Name         string              `json:"name"`
+	Count        int                 `json:"count"`
+	LastError    *ErrorStatus        `json:"last_error,omitempty"`
+	RefreshToken *RefreshTokenStatus `json:"refresh_token,omitempty"`
+}
+
+// AuthTokenStatus contains the status of an auth token tracked in the cache for
+// a specific user.
+type AuthTokenStatus struct {
+	Id                    string `json:"id"`
+	KeyringReferences     int    `json:"keyring_references,omitempty"`
+	KeyringlessReferences int    `json:"keyringless_references,omitempty"`
+}
+
+// UserStatus contains the status of a specific user tracked by the cache
+type UserStatus struct {
+	// The Id of the user this status is for
+	Id string `json:"id"`
+	// The boundary address for this user
+	Address string `json:"address"`
+	// The auth tokens used by this user to authenticate with the boundary instance
```

Comment:
Curious in this file generally why some things are omitempty and some aren't. Is it that the ones not marked omitempty are never expected to be optional?